src/System.ComponentModel.Composition/src/System/ComponentModel/Composition/ConstraintServices.cs

```diff
@@ -85,8 +94,15 @@ private static Expression CreateMetadataConstraintBody(IEnumerable<KeyValuePair<
 
         private static Expression CreateCreationPolicyContraint(CreationPolicy policy, ParameterExpression parameter)
         {
-            Assumes.IsTrue(policy != CreationPolicy.Any);
-            Assumes.NotNull(parameter);
+            if(policy != CreationPolicy.Any)
+            {
+                throw new Exception();
```

Comment:
please pass SR.Diagnostic_InternalExceptionMessage to this and other empty Exceptions.